tools/pd-api-bench/config/config.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"github.com/BurntSushi/toml"
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	flag "github.com/spf13/pflag"
+	"github.com/tikv/pd/pkg/utils/configutil"
+	"github.com/tikv/pd/tools/pd-api-bench/cases"
+	"go.uber.org/zap"
+)
+
+// Config is the heartbeat-bench configuration.
+type Config struct {
+	flagSet    *flag.FlagSet
+	configFile string
+	PDAddr     string `toml:"pd" json:"pd"`
+
+	Log      log.Config `toml:"log" json:"log"`
+	Logger   *zap.Logger
+	LogProps *log.ZapProperties
+
+	Client int64 `toml:"client" json:"client"`
+
+	// tls
+	CaPath   string `toml:"ca-path" json:"ca-path"`
+	CertPath string `toml:"cert-path" json:"cert-path"`
+	KeyPath  string `toml:"key-path" json:"key-path"`
+
+	// only for init
+	HTTP map[string]caseConfig `toml:"http" json:"http"`
+	GRPC map[string]caseConfig `toml:"grpc" json:"grpc"`
+}
+
+type caseConfig struct {
+	QPS   int64 `toml:"qps" json:"qps"`
+	Burst int64 `toml:"burst" json:"burst"`
+}
+
+// NewConfig return a set of settings.
+func NewConfig(flagSet *flag.FlagSet) *Config {
+	cfg := &Config{}
+	cfg.flagSet = flagSet
+	fs := cfg.flagSet
+	fs.StringVar(&cfg.configFile, "config", "", "config file")
+	fs.StringVar(&cfg.PDAddr, "pd", "http://127.0.0.1:2379", "pd address")
```

Comment:
How about also supporting the status addr?

Reply:
see #7692